config/plugin/network_problem.sh

```diff
@@ -1,23 +1,35 @@
 #!/bin/bash
 
 # This plugin checks for common network issues. Currently, it only checks
-# if the conntrack table is full. 
+# if the conntrack table is full.
 
 OK=0
 NONOK=1
 UNKNOWN=2
 
-[ -f /proc/sys/net/ipv4/netfilter/ip_conntrack_max ] || exit $UNKNOWN
-[ -f /proc/sys/net/ipv4/netfilter/ip_conntrack_count ] || exit $UNKNOWN
+# "nf_conntrack" replaces "ip_conntrack" - support both
+NF_CONNTRACK_COUNT='/proc/sys/net/netfilter/nf_conntrack_count'
+NF_CONNTRACK_MAX='/proc/sys/net/netfilter/nf_conntrack_max'
+IP_CONNTRACK_COUNT='/proc/sys/net/ipv4/netfilter/ip_conntrack_count'
+IP_CONNTRACK_MAX='/proc/sys/net/ipv4/netfilter/ip_conntrack_max'
 
-conntrack_max=$(cat /proc/sys/net/ipv4/netfilter/ip_conntrack_max)
-conntrack_count=$(cat /proc/sys/net/ipv4/netfilter/ip_conntrack_count)
+if [ -f $NF_CONNTRACK_COUNT ] && [ -f $NF_CONNTRACK_MAX ]; then
+    CONNTRACK_COUNT=$NF_CONNTRACK_COUNT
+    CONNTRACK_MAX=$NF_CONNTRACK_MAX
+elif [ -f $IP_CONNTRACK_COUNT ] && [ -f $IP_CONNTRACK_MAX ]; then
+    CONNTRACK_COUNT=$IP_CONNTRACK_COUNT
+    CONNTRACK_MAX=$IP_CONNTRACK_MAX
+else
+    exit $UNKNOWN
+fi
+
+conntrack_count=$(< $CONNTRACK_COUNT) || exit $UNKNOWN
+conntrack_max=$(< $CONNTRACK_MAX) || exit $UNKNOWN
 
-if (( conntrack_count >= conntrack_max )); then
-    echo "Conntrack table full"
+if (( conntrack_count > conntrack_max /10 * 9)); then
+    echo "Conntrack table is more than 90% full"
```

Comment:
Maybe
`echo "Conntrack table is more than 90% used: ${conntrack_count} out of ${conntrack_max}"`

What do you think?

Reply:
Update to included this additional information for both `OK` and `NONOK` exit.
In addition I choose to output to `STDERR` for `NONOK`

Reply:
Sorry for the trouble. Could you help change the print back to `STDOUT`? :P

This is a small detail in `custom-plugin-monitor`'s interface, where only `STDOUT` is collected, and `STDERR` is ignored. See the [design](https://docs.google.com/document/d/1jK_5YloSYtboj-DtfjmYKxfNnUxCAvohLnsH5aGCAYQ/edit#) and the [implementation](https://github.com/kubernetes/node-problem-detector/blob/8704ec0c42b27db5eca8e57c3dbfc172ba82765d/pkg/custompluginmonitor/plugin/plugin.go#L131) for this detail.